sys/dev_uart.c

```diff
@@ -22,12 +22,28 @@ static int dev_uart_write(vnode_t *t, uio_t *uio) {
 }
 
 static int dev_uart_read(vnode_t *t, uio_t *uio) {
+  char buffer[UART_BUF_MAX];
+  unsigned curr = 0;
+  while (curr < UART_BUF_MAX && curr < uio->uio_resid) {
+    buffer[curr] = uart_getchar();
```

Comment:
Ok. I'm currently working on intr-sketch branch to implement basic functionality for handling interrupts.